### ISSUE TITLE
DeclarePublicAPI doesn't handle VB autoproperties

### ISSUE THREAD

**Issue by Pilchie** [dotnet/roslyn-analyzers#1082]
#### Analyzer package

Roslyn.Diagnostics.Analyzers

#### Analyzer

Example: DeclarePublicAPIAnalyzer

#### Repro steps

```VB
Public Class C
    Public Property P As Integer
End Class
```
#### Expected behavior
Should report this as public API

#### Actual behavior
It does not

#### NOTES
This happens because we only look at methods, and the analyzer engine doesn't pass us accessors that are `IsImplicitlyDeclared`.

**Comment by mavasani** [comment]
I am sure there is a dupe of this already, can't find it now

**Comment by mavasani** [comment]
The underlying compiler bug was resolved as by design: https://github.com/dotnet/roslyn/issues/11258

We need to identify a way to workaround this in the analyzer.

**Comment by mavasani** [comment]
This has been fixed by @CyrusNajmabadi
